mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -60,6 +63,24 @@ struct GemmRewritePattern : public OpConversionPattern<GemmOp> {
 };
 } // end namespace
 
+static Type obtainAccumulatorElementType(OpBuilder &b, Type elementTypeA,
+                                         Type elementTypeB, Type destType) {
+  // Determine the type used on VGPR to act as accumulator.
+  // f32: f32.
+  // f16, bf16: f32 to prevent overflow from happening.
+  // i16 : i16.
+  // fp8 (any combo) : f32.
+  // i8: i32, since we have an i32 output
+  Type accumulatorType = destType;
+  if (elementTypeA.isF16() || elementTypeA.isBF16() ||
```

Comment:
This could be `isa<FloatType> && elementType.getIntOrFloatBitwidth() < 32` or something like that.

Reply:
Thanks, it looks much better.
Done